src/hotspot/share/classfile/resolutionErrors.cpp

```diff
@@ -127,10 +127,7 @@ ResolutionErrorEntry::~ResolutionErrorEntry() {
 }
 
 void ResolutionErrorEntry::set_nest_host_error(const char* message) {
-  // If a message is already set, free it.
-  if (nest_host_error() != nullptr) {
-    FREE_C_HEAP_ARRAY(char, _nest_host_error);
-  }
+  assert(_nest_host_error == nullptr, "caller should have checked");
```

Comment:
May also check that you're holding `SystemDictionary_lock`? That way you're more certain that the field has not been changed since the caller checked it.